src/main/java/com/google/flourbot/datastorage/FirebaseDataStorage.java

```diff
@@ -59,7 +59,7 @@ public FirebaseDataStorage() {
       document = querySnapshot.get().getDocuments().get(0);
       return document.exists() ? Optional.of(document) : Optional.empty();
     } catch (IndexOutOfBoundsException e) {
-      throw new IllegalStateException(e);
+      return Optional.empty();
```

Comment:
Oh, if IndexOutOfBoundsException is possible, we should check isEmpty first before .get(0) and return empty optional if that's the case. Exceptions are expensive and complicated, if we can we should avoid them.

Reply:
Oh I see. I don't see a .emtpy() method on `DocumentSnapshot`  type though.